python/hail/expr/aggregators.py

```diff
@@ -998,6 +998,12 @@ def hist(expr, start, end, bins) -> StructExpression:
     return _agg_func('hist', expr, t, start, end, bins)
 
 
+@typecheck(x=expr_float64, y=expr_float64,
+           xmin=expr_float64, xmax=expr_float64, ymin=expr_float64, ymax=expr_float64)
```

Comment:
I know you copied this from `hist`. But I'm thinking the type checks should just be `float` and not an expr. This is because they are constructor arguments to the aggregator and shouldn't vary by row or column. Or they can be an expr but you have to check the axes are the empty set. You can get the expression source and axes from `._indices`. There might be helper methods to do this already or you can add one.

You also need to check that `x` and `y` are from the same source. Look at `_indices` again and also if there are helper functions for this as well. Maybe `analyze`?